plaid-stl/src/npm/mod.rs

```diff
@@ -0,0 +1,607 @@
+        // TODO We could check some things about name and description. E.g., the description
+        // should be longer than 8 characters to avoid "test" and "testing".
+        Self {
+            token_name: token_name.to_string(),
+            token_description: token_description.to_string(),
+            allowed_ip_ranges: None,
+            expiration_days: None,
+            orgs_permission: None,
+            packages_and_scopes_permission: None,
+            selected_orgs: None,
+            selected_packages: None,
+            selected_packages_and_scopes: None,
+            selected_scopes: None,
+        }
+    }
+}
+
+/// An npm token configured on the npm website for the current user
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[allow(dead_code)]
+pub struct NpmToken {
+    token: String,
+    token_name: Option<String>,
+    pub token_type: Option<String>,
+    // Deserialize the date field from the ISO 8601 format
+    #[serde(default, deserialize_with = "deserialize_option_timestamp")]
+    expires: Option<DateTime<Utc>>,
+}
+
+// Custom deserializer for an optional DateTime<Utc>
+fn deserialize_option_timestamp<'de, D>(deserializer: D) -> Result<Option<DateTime<Utc>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    // Try to deserialize as a string (ISO 8601 format) or return None if missing
+    let opt = Option::<String>::deserialize(deserializer)?;
+
+    // Parse the timestamp string into a DateTime<Utc>, if it's present
+    Ok(opt.and_then(|s| {
+        DateTime::parse_from_rfc3339(&s)
+            .ok()
+            .map(|dt| dt.with_timezone(&Utc))
+    }))
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct SetTeamPermissionOnPackageParams {
+    pub team: String,
+    pub package: String,
+    pub permission: NpmPackagePermission,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct CreateGranularTokenForPackageParams {
+    pub package: String,
+    pub specs: GranularTokenSpecs,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct AddRemoveUserToFromTeamParams {
+    pub user: String,
+    pub team: NpmTeam,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct InviteUserToOrganizationParams {
+    pub user: String,
+    pub team: Option<NpmTeam>,
+}
+
+/// Access level for an npm package
+#[derive(Serialize, Deserialize)]
+pub enum PkgAccessLevel {
+    Restricted,
+    Public,
+}
+
+impl PkgAccessLevel {
+    pub fn to_string(&self) -> String {
+        match self {
+            PkgAccessLevel::Restricted => "restricted".to_string(),
+            PkgAccessLevel::Public => "public".to_string(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct PublishEmptyStubParams {
+    pub package_name: String,
+    pub access_level: PkgAccessLevel,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ListPackagesWithTeamPermissionParams {
+    pub team: NpmTeam,
+    pub permission: NpmPackagePermission,
+}
+
+// End of "All the structs which are also used in the Plaid API"
```

Comment:
If this is a meaningful distinction, I would move them to another file so they can be in their own mod. That'll also make it a lot more clear when updating

Reply:
Good idea, moved to separate module in the STL.